addons/website_payment/static/src/snippets/s_donation/donation_snippet.js

```diff
@@ -0,0 +1,198 @@
+
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+import { formatCurrency } from "@web/core/currency";
+import { rpc } from "@web/core/network/rpc";
+import { _t } from "@web/core/l10n/translation";
+
+const CUSTOM_BUTTON_EXTRA_WIDTH = 10;
+let cachedCurrency;
+
+export class DonationSnippet extends Interaction {
+    static selector = ".s_donation";
+    dynamicContent = {
+        ".s_donation_btn": { "t-on-click": this.onClickPrefilled },
+        ".s_donation_donate_btn": { "t-on-click": this.onClickDonate },
+        "#s_donation_range_slider": { "t-on-input": this.onInputRangeSlider },
+    };
+
+    setup() {
+        this.rangeSlider = this.el.querySelector('#s_donation_range_slider');
+        this.defaultAmount = this.el.dataset.defaultAmount;
+        if (!!this.rangeSlider) {
+            this.rangeSlider.value = this.defaultAmount;
+            this.setBubble(this.rangeSlider);
+        }
+    }
+
+    async willStart() {
+        await this.displayCurrencies();
```

Comment:
cf my other comment, maybe you can just store `this.currency = await this.getCurrency()` and then use `this.currency` to do the rest in an interaction-friendly way (I haven't checked in details how to do it, as I said it just looks like a good candidate)